### DISCUSSION TITLE
TH16 con sensor si7020

### DISCUSSION THREAD

**Discussion by Danilo0609** [RavenSystem/esp-homekit-devices#1616] [Q&A]
HOla!!! flashes sin problemas con el haa excelente.
no puedo usar la automatización con la temperatura. solo con el horario. se puede hacer algo con esto? necesito automatizar una caldera acorde a la temperatura pero no puedo. me podrán ayudar?

**Comment by xrust83** [comment]
https://github.com/RavenSystem/esp-homekit-devices/wiki/HeaterCooler

**Reply by Danilo0609** [reply]
mmmmm. no entiendo mucho
te comento:
fashee mi th 16 con sensor Si7021 con HAA fullaboot.
luego coloque el siguiente JSON: 
![image](https://user-images.githubusercontent.com/93847265/140765928-6d8fad41-8bce-4fe2-b3b3-ff022b2d1322.png)
espere los 10 minutos y agregue el dispositivo a la aplicacion casa. 
y alli no me permite realizar la automatizacion de la caldera con la temperatura ( no me permite encender o apagar la caldera con respecto a la temperatura). si con la hora. necesito automatizar con respecto a la temperatura. me podras guiar? gracias

**Reply by xrust83** [reply]
¿Necesitas json o ya has agregado el dispositivo?  tomar una captura de pantalla de homekit

**Reply by Danilo0609** [reply]
Creo que el JSON que utilice es erróneo. Me permite acceder al th16 puedo ver la temperatura y la humedad pero no puedo automatizar el sensor con respecto a la temperatura

**Comment by xrust83** [comment]
https://github.com/RavenSystem/esp-homekit-devices/issues/891

**Comment by xrust83** [comment]
https://github.com/RavenSystem/esp-homekit-devices/wiki/Temperature-and-Humidity-Sensors#sensor-type

For si7021 "n"=4

**Reply by Danilo0609** [reply]
Si eso está bien.

**Reply by Danilo0609** [reply]
Necesito que la caldera se apague a 24 grados y se encienda a 20 grados con dead band de 4 grados. Cómo sería el JSON?

**Reply by Danilo0609** [reply]
Muchísimas gracias por la respuesta. He podido solucionar parte del problema con estos json.

EL json que utilice es


{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"0":{"r":[{"g":12}]},"1":{"a":0},"3":{"r":[{"g":12,"v":1}]},"5":{"a":0},"t":21,"s":5,"g":14,"b":[],"n":4,"j":10,"w":1,"m":10,"x":40},{"0":{"r":[{"g":12,"v":0}]},"1":{"r":[{"g":12,"v":1}]},"t":1,"b":[{"g":0,"t":1}],"s":0},{"t":24,"b":[],"n":4,"g":14,"j":10}]}


Pero no este json no puedo controlar el dispositivo con respecto a la humedad y tampoco  no puedo establecer un rango de grados entre el encendido y el apagado del th16 como con el firmware original de sonoff. Se podrá configurar un json completo?. 

Nueva consulta. Como reseteo el wifi de un Device ya instalado para cambiarlo? Con el boton fisico no me lo permitió. Lo presione durante 8 segundos y no me deja conectarme  A LA RED HAA. Da error. No se como resetearlo. 

Muchísimas gracias. Excelente post!!!!!!!

**Reply by Danilo0609** [reply]
Hola como estas. logre hacerlo funcionar a medias. utilice este json
![image](https://user-images.githubusercontent.com/93847265/141333090-7aa8ae88-9d52-440f-aa9c-c921f2daa844.png)
pero el boton de encendido no funciona y no responde al rango de temperatura. me podran ayudar a reprogramar el json
![IMG_6023](https://user-images.githubusercontent.com/93847265/141334195-78582498-5f42-499f-aa35-df4c395235ef.PNG)
LO PROGRAME PARA ZONA DE CONFORT ENTRE 27 Y 28 GRADOS
![IMG_6021](https://user-images.githubusercontent.com/93847265/141334426-f0d64dd8-bc59-45a7-8f60-e29e389a2d6b.jpg)
A LOS 27 GRADOS SE ENCIENDE UN SOLO BOTON (TERMOSTATO) PERO EL SWITCH NO SE ENCIENDE. SI AGRUPO LOS DOS SWITCH SI LOGRO QUE SE ENCIENDA.
![IMG_6022](https://user-images.githubusercontent.com/93847265/141334298-32e3a396-ad5c-4c3a-9120-965c357697e7.jpg)
PERO CUANDO SUPERA LA TEMPERATURA MAYOR DESEADA NO SE APGAN NINGUNO DE LOS SWITCHS. SEGURAMENTE ES UYN ERROR DE JSON.

HELP ME PLEASE!!!!

**Reply by Danilo0609** [reply]
MUCHAS GRACIAS X TODO CHICOS. LOGRE QUE FUNCIONE PERFECTO!!!!!!!!!
LES DEJO EL JSON PARA COMPARTIR. LO UTILICE CAMBIANDO LA BANDA MUERTA A d:0 Y AMPLIANDO LA HUMEDAD DE 0 A 100% ( NO DESEO CONTROL X HUMEDAD EN ESTE CASO.

**Comment by Danilo0609** [comment]
> https://github.com/RavenSystem/esp-homekit-devices/wiki/HeaterCooler
